tachyon/zk/plonk/halo2/proof_reader.h

```diff
@@ -54,15 +54,17 @@ class ProofReader {
         verifying_key_.constraint_system();
     proof_.advices_commitments_vec.resize(num_circuits_);
     for (size_t i = 0; i < num_circuits_; ++i) {
-      proof_.advices_commitments_vec[i].reserve(
+      proof_.advices_commitments_vec[i].resize(
           constraint_system.num_advice_columns());
     }
     proof_.challenges.reserve(constraint_system.challenge_phases().size());
     for (Phase current_phase : constraint_system.GetPhases()) {
       for (size_t i = 0; i < num_circuits_; ++i) {
-        for (Phase phase : constraint_system.advice_column_phases()) {
-          if (current_phase == phase) {
-            proof_.advices_commitments_vec[i].push_back(Read<C>());
+        const std::vector<Phase>& advice_column_phases =
+            constraint_system.advice_column_phases();
+        for (size_t j = 0; j < advice_column_phases.size(); ++j) {
+          if (current_phase == advice_column_phases[j]) {
+            proof_.advices_commitments_vec[i][j] = Read<C>();
           }
         }
       }
```

Comment:
```suggestion
    size_t num_advice_columns = constraint_system.num_advice_columns();
    for (size_t i = 0; i < num_circuits_; ++i) {
      proof_.advices_commitments_vec[i].resize(num_advice_columns);
    }
    proof_.challenges.reserve(constraint_system.challenge_phases().size());
    for (Phase current_phase : constraint_system.GetPhases()) {
      for (size_t i = 0; i < num_circuits_; ++i) {
        const std::vector<Phase>& advice_column_phases =
            constraint_system.advice_column_phases();
        for (size_t j = 0; j < num_advice_columns; ++j) {
          if (current_phase == advice_column_phases[j]) {
            proof_.advices_commitments_vec[i][j] = Read<C>();
          }
        }
      }
```
How about like this since we've resized `advice_commitments_vec` with `num_advice_columns`?

Reply:
I don't notice a significant distinction. Could you please clarify the advantage of using `num_advice_columns`?

Reply:
```c++
        for (size_t j = 0; j < num_advice_columns; ++j) {
```
@chokobole It's for this line, where we iterate over `num_advice_columns`. It seems like personal prefernce, but I just thought this is more readable since we resized `advice_commitments_vec` with `num_advice_columns`. You may not apply it since it seems like personal preference.